### PR TITLE
COMMERCE-7935 Add short description to product card and expose diagram image method to helper

### PR DESCRIPTION
**Pull request by dsitu**
https://issues.liferay.com/browse/COMMERCE-7935

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:stop

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Review by ivicac — APPROVED** [review]

File: modules/apps/commerce/commerce-shop-by-diagram-api/src/main/java/com/liferay/commerce/shop/by/diagram/util/CSDiagramCPTypeHelper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	public CSDiagramSetting getCSDiagramSetting(
+			CPRequestHelper cpRequestHelper, long cpDefinitionId)
+		throws PortalException;
+
+	public CSDiagramType getCSDiagramType(String type);
+
+	public String getImageURL(
+			CPRequestHelper cpRequestHelper, long cpDefinitionId)
+		throws Exception;
```
[the remark this answers is not in the document]
Comment (dsitu) [reply modules/apps/commerce/commerce-shop-by-diagram-api/src/main/java/com/liferay/commerce/shop/by/diagram/util/CSDiagramCPTypeHelper.java]: @ivicac sorted method arguments and resolved merge conflict, thanks for catching this.

**Comment by ivicac** [comment]
ci:forward

**Comment by ivicac** [comment]
ci:test:relevant

**Comment by ivicac** [comment]
ci:forward

**Review by ivicac — APPROVED** [review]

**Comment by ivicac** [comment]
ci:reopen

File: modules/dxp/apps/commerce/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/render/list/entry/view.jsp [hunk 2]
```diff
@@ -46,6 +46,12 @@ CSDiagramCPTypeDisplayContext csDiagramCPTypeDisplayContext = (CSDiagramCPTypeDi
 						</span>
 					</a>
 				</p>
+
+				<p class="card-text" title="<%= cpCatalogEntry.getShortDescription() %>">
+					<span class="text-truncate-inline">
+						<span class="text-truncate"><%= cpCatalogEntry.getShortDescription() %></span>
+					</span>
+				</p>
 			</div>
```
Comment (ivicac) [inline modules/dxp/apps/commerce/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/render/list/entry/view.jsp:55]: hey @dsitu we don't need this. We will leave customer make customization by implementing their custom renderer.

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
@ivicac i've removed the short description from the product card, this is ready to be reviewed again.

**Comment by ivicac** [comment]
hey @dsitu to allow csutomer to write custom renderer we need to expose additional classes. Can you please check https://issues.liferay.com/browse/COMMERCE-7935 and also /commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/render/list/view.jsp what classes are necessary to export in order to make custom view.jsp?
According to Giovanni

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
@ivicac I've completely removed `com.liferay.commerce.shop.by.diagram.admin.web.internal.display.context.CSDiagramCPTypeDisplayContext` and both view.jsp's are now using `com.liferay.commerce.shop.by.diagram.util.CSDiagramCPTypeHelper`. I don't think we need to expose `com.liferay.commerce.shop.by.diagram.admin.web.internal.display.context.CSDiagramSettingDisplayContext` and `com.liferay.commerce.shop.by.diagram.admin.web.internal.type.DefaultCSDiagramType`. They seem to only be used for `edit_cs_diagram_setting.jsp` which is in the control panel pages. Let me know if we do need to expose them or anything else, i'm not really familiar with how custom renders work. Thanks!

**Comment by ivicac** [comment]
ci:forward

**Pull request closed without merging**